builder/builder-next/control/control.go

```diff
@@ -256,9 +274,35 @@ func (c *Controller) Solve(ctx context.Context, req *controlapi.SolveRequest) (*
 		return nil, err
 	}
 	if req.Exporter != "" {
-		exp, err := w.Exporter(req.Exporter, c.opt.SessionManager)
-		if err != nil {
-			return nil, err
+		var exp exporter.Exporter
+		if c.opt.UseSnapshotter {
+			if req.Exporter == "moby" {
```

Comment:
Yes, agree; I'm also looking at this, and wondering;

- should this be using `client.ExporterDocker` ? https://github.com/moby/moby/blob/6eab4f55faf77bb0f253553d7d6fadb2ef6f4669/vendor/github.com/moby/buildkit/client/exporters.go#L8
- it looks like we're using two different approaches; this one (update the value _before_ calling `worker.Exporter()`), and `ContainerdWorker.Exporter()`, which does so internally. We should pick one.

Reply:
* The docker exporter is to export a [docker tarball](https://github.com/moby/buildkit#docker-tarball)
* we are doing this in two different ways because `DOCKER_BUILDKIT=1 docker build` and `docker buildx build` are two different things